app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -61,15 +100,15 @@ class PromotedStoryListView @JvmOverloads constructor(
     if (adapter == null) {
       adapter = createAdapter()
     }
-    if (newDataList == null) {
-      oppiaLogger.w(PROMOTED_STORY_LIST_VIEW_TAG, "Failed to resolve new story list data")
+    if (::promotedDataList.isInitialized) {
```

Comment:
Isn't this guaranteed now since the method is only called after everything is initialized? It seems like we can simplify this code, then, as the error case shouldn't happen anymore.

Reply:
Yeah Ben it is guaranteed now. Updating.